api/comments/serializers.py

```diff
@@ -0,0 +1,146 @@
+    date_modified = ser.DateTimeField(read_only=True)
+    modified = ser.BooleanField(read_only=True, default=False)
+    deleted = ser.BooleanField(read_only=True, source='is_deleted', default=False)
+
+    # LinksField.to_representation adds link to "self"
+    links = LinksField({})
+
+    class Meta:
+        type_ = 'comments'
+
+    def create(self, validated_data):
+        user = validated_data['user']
+        node = validated_data['node']
+
+        validated_data['content'] = validated_data.pop('return_content')
+        if node and node.can_comment(Auth(user)):
+            comment = Comment.create(auth=Auth(user), **validated_data)
+        else:
+            raise PermissionDenied("Not authorized to comment on this project.")
+        return comment
+
+    def update(self, comment, validated_data):
+        assert isinstance(comment, Comment), 'comment must be a Comment'
+        auth = Auth(self.context['request'].user)
+        if validated_data:
+            if 'return_content' in validated_data:
+                comment.edit(validated_data['return_content'], auth=auth, save=True)
+            is_deleted = validated_data.get('is_deleted', None)
+            if is_deleted:
+                comment.delete(auth, save=True)
+            elif comment.is_deleted:
+                comment.undelete(auth, save=True)
+        return comment
+
+    def get_target_type(self, obj):
+        target_id = obj._id
+        guid_object = Guid.load(target_id)
+        if not guid_object:
+            raise NotFound('Comment target not found.')
+        target = guid_object.referent
+
+        if isinstance(target, Node):
+            return 'node'
+        elif isinstance(target, Comment):
+            return 'comment'
+        else:
+            raise InvalidModelValueError('Invalid comment target.')
+
+
+class CommentDetailSerializer(CommentSerializer):
+    """
+    Overrides CommentSerializer to make id required.
+    """
+    id = IDField(source='_id', required=True)
+    deleted = ser.BooleanField(source='is_deleted', required=True)
+
+
+class CommentReportsSerializer(JSONAPISerializer):
+    id = IDField(source='_id', read_only=True)
+    type = TypeField()
+    category = ser.ChoiceField(choices=[('spam', 'Spam or advertising'),
+                                        ('hate', 'Hate speech'),
+                                        ('violence', 'Violence or harmful behavior')], required=True)
+    message = ser.CharField(source='text', required=False, allow_blank=True)
+    links = LinksField({'self': 'get_absolute_url'})
+
+    class Meta:
+        type_ = 'comment_reports'
+
+    def get_absolute_url(self, obj):
+        comment_id = self.context['request'].parser_context['kwargs']['comment_id']
+        return absolute_reverse(
+            'comments:report-detail',
+            kwargs={
+                'comment_id': comment_id,
+                'user_id': obj._id
+            }
+        )
+
+    def create(self, validated_data):
```

Comment:
I commented about the comment report class above: 

>This class is just for the purpose of serializing comment report data into a format that is more compatible with the API, since that information is in the Comment model's reports field and not it's own model.

If that doesn't matter, then I can go ahead and make the `create` and `update` class methods.